src/library_pthread.js

```diff
@@ -1052,6 +1052,8 @@ var LibraryPThread = {
   _emscripten_notify_proxying_queue: function(targetThreadId, currThreadId, mainThreadId, queue) {
     if (targetThreadId == currThreadId) {
       setTimeout(() => {
+        // Only execute the queue if we have a live pthread runtime. We
+        // implement pthread_self to return 0 if there is no live runtime.
```

Comment:
I think there is probably better check for "is runtime alive" than this.. `callUserCallback` does this for you.. which will be used if you use `safeSetTimeout`. 

The condition used in `callUserCallback` is:

```
#if EXIT_RUNTIME                                                                    
    if (runtimeExited || ABORT) {                                                   
#else                                                                                                                          
    if (ABORT) {                                                                 
#endif    
```

We could factor that out into `isRuntimeAlive()`.. or you can just use `callUserCallback` here.

Reply:
Sure, I can change to use `callUserCallback` here.

Reply:
I guess if I do that, this comment won't be necessary. I'll make a new PR, then.